mqt/bench/benchmark_generator.py

```diff
@@ -580,22 +587,28 @@ def get_one_benchmark(
 
     elif level == "mapped" or level == 3:
         if compiler == "qiskit":
+            opt_level = compiler_settings["qiskit"]["optimization_level"]
             qc_mapped = qiskit_helper.get_mapped_level(
                 qc,
                 gate_set_name,
                 circuit_size,
                 device_name,
-                compiler_settings,
+                opt_level,
                 False,
                 True,
             )
         elif compiler == "tket":
+            placement = compiler_settings["tket"]["placement"].lower()
+            if placement == "lineplacement":
+                lineplacement = True
+            else:
+                lineplacement = False
             qc_mapped = tket_helper.get_mapped_level(
                 qc,
                 gate_set_name,
                 circuit_size,
                 device_name,
-                compiler_settings,
+                lineplacement,
```

Comment:
In the future, i.e., when there are more options for the respective compilers, this interface will probably need a rework.
I believe the best solution here would be to pass the `compiler_settings["tket"]` dictionary (same for qiskit).
In that fashion, all the handling of compiler options is done in the `*_helper` classes and not sprinkled throughout here.

I'd understand, if you don't want to change this right now since it's at least a medium effort, so feel free to decide whether to tackle this right now or at a later point in time.

Reply:
That is indeed a very good suggestion.

I will open another issue to tackle that problem in the future.